### PR TITLE
Add nodeletctl

### PR DESCRIPTION
**Pull request by xagent003**
Moving everything over from airctl to nodelet repo.

### CODE REVIEW

**Comment by xagent003** [comment]
I have kept the ./airctl advanced-ddu commands - they call directly into the ndoeletctl Golang pkg code from here. But I've also added nearly same ./nodeletctl cobra CMDs as well.

File: nodeletctl/cmd/create.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		if err != nil {
+			fmt.Printf("\nFailed to create nodelet cluster: %s\n", err)
+		}
+		return err
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(createClusterCmd)
+	createClusterCmd.Flags().StringVar(&clusterBootstrapFile, "config", "/root/nodeletCluster.yaml", "Path to nodelet bootstrap config")
```
Comment (roopakparikh) [inline nodeletctl/cmd/create.go:25]: suggestion: we use $HOME/nodeletCluster.yaml instead of /root

File: nodeletctl/cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+func init() {
+
+	cobra.OnInitialize(initConfig)
+
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+
+	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/airctl-config.yaml)")
```
Comment (roopakparikh) [inline nodeletctl/cmd/root.go:51]: airctl-config ? shouldn't that be nodelet-config ?

File: nodeletctl/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// Find home directory.
+		home, err := homedir.Dir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".airctl" (without extension).
```
Comment (roopakparikh) [inline nodeletctl/cmd/root.go:75]: .airctl ? - please remove references to airctl in many places

File: nodeletctl/cmd/scale.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+		if err != nil {
+			fmt.Printf("\nFailed to update nodelet cluster: %s\n", err)
+		}
+		return err
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(scaleClusterCmd)
+	scaleClusterCmd.Flags().StringVar(&clusterBootstrapFile, "config", "/root/nodeletCluster.yaml", "Path to nodelet bootstrap config")
```
Comment (roopakparikh) [inline nodeletctl/cmd/scale.go:25]: config file location: lets use a better place , I suggesting using $HOME/nodeletCluster or /etc/pf9 ...

File: nodeletctl/go.mod [hunk 5]
```diff
@@ -0,0 +1,23 @@
+module nodeletctl
+
+replace (
+	github.com/platform9/nodelet/nodeletctl => /home/xagent/nodelet/nodeletctl
```
Comment (roopakparikh) [inline nodeletctl/go.mod:4]: assuming we will be replacing this going forward

**Review by roopakparikh — COMMENTED** [review]
Minor comments, I am assuming most of the code is just moved from the airctl to nodeletctl ?

**Comment by roopakparikh** [comment]
@xagent003 can you also add a README

File: nodeletctl/cmd/create.go (hunk 1 above)
Comment (xagent003) [reply nodeletctl/cmd/create.go]: Fixed

File: nodeletctl/cmd/root.go (hunk 2 above)
Comment (xagent003) [reply nodeletctl/cmd/root.go]: Fixed

**Pull request merged**